api/src/middlewares/services/UserService/personalisation/personalisationBuilder.userConfirmation.ts

```diff
@@ -4,17 +4,29 @@ import { AnswersHashMap } from "../../../../types/AnswersHashMap";
 import { FormType, PayMetadata } from "../../../../types/FormDataBody";
 import { ApplicationError } from "../../../../ApplicationError";
 
-const previousMarriageDocs = {
-  Divorced: "decree absolute",
-  "Dissolved civil partner": "final order",
-  Widowed: "late partner's death certificate",
-  "Surviving civil partner": "late partner's death certificate",
-  Annulled: "decree of nullity",
+type PersonalisationFunction = (fields: AnswersHashMap) => Record<string, boolean>;
+
+const personalisationTypeMap: Record<FormType, PersonalisationFunction> = {
+  affirmation: getAffirmationPersonalisations,
+  cni: getCNIPersonalisations,
+  exchange: (_fields: AnswersHashMap) => ({}),
+  msc: (_fields: AnswersHashMap) => ({}),
+  cniAndMsc: (_fields: AnswersHashMap) => ({}),
 };
 
 export function buildUserConfirmationPersonalisation(answers: AnswersHashMap, metadata: { reference: string; payment?: PayMetadata; type?: FormType }) {
   const isSuccessfulPayment = metadata.payment?.state?.status === "success" ?? false;
-  const docsList = buildUserConfirmationDocsList(answers, metadata.type);
+
+  if (!answers) {
+    throw new ApplicationError("WEBHOOK", "VALIDATION", 500, "Fields are empty");
+  }
+
+  const getAdditionalPersonalisations = personalisationTypeMap[metadata.type as string];
```

Comment:
we shouldn't cast to a string here. it's already a typeof FormType, attempting to access a record keyed with FormType. The type error without the cast comes from the fact that metadata.type might be undefined. 

we can just force unwrap with `!`, because we handle what happens when metadata.type is undefined (or an unknown value) in l25-27.

```suggestion
  const getAdditionalPersonalisations = personalisationTypeMap[metadata.type!];
```